Pendulum_with_elastic_balls.rst

```diff
@@ -0,0 +1,540 @@
+.. code:: ipython3
+
+    import sympy as sm
+    import sympy.physics.mechanics as me
+    import time
+    import numpy as np
+    from scipy.integrate import odeint, solve_ivp
+    from scipy.optimize import fsolve, minimize
+    import matplotlib.pyplot as plt
+    import matplotlib
+    %matplotlib inline
+    
+    import pythreejs as p3js
+
+
+.. jupiter-execute::
+
+    '''
+    A pendulum consisting on n massless rods of length l. A ball of mass m and radius r is attached to each
+    rod, such that the rod goes through the center of each ball. the center of the ball is fixed at the middle
+    of the rod. the ball rotates around the rod.
+    If two balls collide, they are ideally elastic, with 'spring constant' k.
+    The collision is modeled using the sm.Heaviside(..) function.
+    The balls are ideally slick, so collisions will not affect their rotation.
+    
+    The total energy does not always remain constant.
+    As method = 'Radau' in solve_ivp gives much 'better' results than no method (e.g. 15% deviation vs. 0.3% in 
+    some cases), I assume that this is due to numerical errors in the integration.
+    - of course I do not know for sure.
+    
+    
+    cse = True reduced the time it take to integrate substantially, by a factor of over 50
```

Comment:
Move any test like this outside of the code cells (outside of `.. jupyter-execute::`.